add handler to inline template call

leejo opened the pull request:
add template_format plugin config option

in the case of using alternate template rendering engines where the
app->renderer->default_handler is set to something other than the
mojo default of 'ep' as this will cause the template markup we are
using in recaptcha_get_html to not be interpolated

template_format will default to 'mojo', i.e. the embedded perl we
are already using, and can be overridden with 'tt' at present to
support Template::Toolkit markup. other markup formats could perhaps
be added

heikojansen commented:
Hi and thanks for your pull request. Sorry it took me so some days to get back to you.
I think I understand the problem and it definitely needs a solution; I am, however, not really sure if what you propose is the way to go.

There aren´t all that many templating engines available to use with Mojolicious but I nevertheless don´t really like to maintain a bunch of template snippets with syntax I´m not familiar with, all of which should generate the exact same output. 
On the other hand, the "EP" template renderer provided as part of Mojolicious is always available - no matter what is used to render the locally defined templates and nobody outside of Mojolicious::Plugin::ReCAPTCHAv2 should care about the template used internally.

So wouldn´t it be sufficient if we simply added the option "handler => 'ep'" to the invocation of "render_to_string" in ReCAPTCHAv2.pm, line 51 (and likewise to "render" in line 58)?
At least that seemed to work in a simple test app I just threw together using Text::XSlate.
Could you give that a try in your app as well?

By the way: there´s a .gitignore file one level above; what was the reason for adding another one?

Best,
Heiko

leejo commented:
Yep, just providing `handler => 'ep'` sounds like a much better solution (and one i wasn't aware of). You can rm the `.gitignore` file. The extra level of directory structure in this repo caught me out when i saw some files not being ignored, i thought i was in the top level of the repo hence didn't see the already existing `.gitignore` a level up.

leejo committed:
add `handler => 'ep'` to inline templates

in the case of using alternate template rendering engines where the
app->renderer->default_handler is set to something other than the
mojo default of 'ep' as this will cause the template markup we are
using in recaptcha_get_html to not be interpolated if the handler
is not passed to the inline template

add test that overrides default_handler to check this, tweak the
.gitignore file to ignore a couple more build related files/dirs

leejo commented:
Ok, this looks good - i've `git reset` and made the suggested change along with adding a small test for the change.
